clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClanEnergyListener.java

```diff
@@ -0,0 +1,66 @@
+package me.mykindos.betterpvp.clans.clans.listeners;
+
+import com.google.inject.Inject;
+import me.mykindos.betterpvp.clans.Clans;
+import me.mykindos.betterpvp.clans.clans.Clan;
+import me.mykindos.betterpvp.clans.clans.ClanManager;
+import me.mykindos.betterpvp.clans.clans.events.EnergyCheckEvent;
+import me.mykindos.betterpvp.core.config.Config;
+import me.mykindos.betterpvp.core.framework.updater.UpdateEvent;
+import me.mykindos.betterpvp.core.gamer.GamerManager;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import me.mykindos.betterpvp.core.utilities.UtilServer;
+import me.mykindos.betterpvp.core.utilities.model.display.TitleComponent;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.title.Title;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+
+import java.util.Optional;
+
+@BPvPListener
+public class ClanEnergyListener extends ClanListener{
+
+    private final Clans clans;
+
+
+    @Inject
+    @Config(path="clans.energy.energyWarnLevel", defaultValue = "1.0")
+    private double energyWarnLevel;
+
+    @Inject
+    ClanEnergyListener(Clans clans, ClanManager clanManager, GamerManager gamerManager) {
+        super(clanManager, gamerManager);
+        this.clans = clans;
+    }
+
+    @UpdateEvent(delay = 30 * 1000, isAsync = true)
+    public void checkEnergy() {
+        Bukkit.getOnlinePlayers().forEach(player -> {
+            Optional<Clan> clanOptional = clanManager.getClanByPlayer(player);
+            if (clanOptional.isPresent()) {
+                Clan clan = clanOptional.get();
+                UtilServer.runTaskLater(clans, () -> {UtilServer.callEvent(new EnergyCheckEvent(player, clan));}, 5);
+            }
+        });
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR)
+    public void onEnergyCheck(EnergyCheckEvent event) {
+        if (event.isCancelled()) return;
+        Player player = event.getPlayer();
+        Clan clan = event.getClan();
+        if (clan.getEnergyRatio() < energyWarnLevel) {
```

Comment:
So with the default configuration, its only going to send the titles if they are below 1 energy?

Reply:
It sends it if it is below 1 hour of energy (obviously subject to change)

Reply:
getEnergyRatio() gets the time in hours (I discovered)

Reply:
I think setting the timer to every 5 minutes should be frequent enough

There will also be a /c shop command by the way, which will let clans buy energy from their base

Reply:
That was the intention, was 30 seconds for testing purposes.

Could add that to the chat component when that exists too.